tests/validation/models.py

```diff
@@ -136,3 +136,52 @@ class Meta:
         constraints = [
             models.UniqueConstraint(Lower('field'), name='func_lower_field_uq'),
         ]
+
+
+class Product(models.Model):
+    price = models.IntegerField(null=True)
+    discounted_price = models.IntegerField(null=True)
+
+    class Meta:
+        required_db_features = {
+            'supports_table_check_constraints',
+        }
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(price__gt=models.F('discounted_price')),
+                name='price_gt_discounted_price_validation',
+            ),
+        ]
+
+
+class ChildProduct(Product):
+    pass
```

Comment:
I think this model is causing the MySQL failures.

Since `Product` cannot be created on MySQL due to not having the `supports_table_check_constraints` feature then `ChildProduct` cannot be created either since MTI requires a foreign key (hence the `Cannot add foreign key constraint` error).

```suggestion
class ChildProduct(Product):
     class Meta:
        required_db_features = {
            'supports_table_check_constraints',
        }
```